sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TableEntityAccessHelper.java

```diff
@@ -50,7 +50,7 @@ public static TableEntity createEntity(Map<String, Object> properties) {
             }
         }
 
-        return creator.get().setProperties(properties);
+        return creator.get().setProperties(TableUtils.adjustOdataPropertyMap(properties));
```

Comment:
L42 - should you be checking properties for null at the method start

Reply:
@joshfree That is not necessary because `adjustOdataPropertyMap` makes use of `computeIfPresent` from `Java.util.Map` which will ignore null properties.